packages/react-form-renderer/src/form-renderer/enhanced-on-change.js

```diff
@@ -40,19 +41,45 @@ const convertType = (dataType, value) => ({
   [dataTypes.BOOLEAN]: castToBoolean(value),
 })[dataType] || value;
 
+/**
+ * Checks the value and returns undefined if its empty. Converst epmtry strings, arrays and objects.
+ * If value is empty its overriden to undefined for further processing.
+ * @param {Any} value Any JS variable to be check if is empty
+ */
+const checkEmpty = value => {
+  if (typeof value === 'number') {
+    return value;
```

Comment:
0 is a valid number, yet, `if (checkEmpty(0))` is falsy. Probably not what you want?

Reply:
Maybe what I'm really saying is that `checkEmpty` should return true/false, and the `typeof checkEmpty(result) === "undefined"` check should just be `checkEmpty(result)`.

Reply:
Is it? There is actually a test with `0` which does return the correct value `0`, not the deleted value.

https://github.com/data-driven-forms/react-forms/pull/273/files#diff-e027b205eabea394b0be42c72f4e7681R49

Maybe I am missing something.

Reply:
Yeah, I did not expect the test to be like that, given the name of the function.
It does work as is, it's just surprising :).

Reply:
Yeah, you might be right. All that 0, -1, etc being false makes me uncomfortable using boolean in this case. Returning boolean directly makes it cleaner.